Sources/GRPCCore/Documentation.docc/Articles/Generating-stubs.md

```diff
@@ -83,3 +91,83 @@ swift build --product protoc-gen-grpc-swift
 
 This command will build the plugin into `.build/debug` directory. You can get the full path using
 `swift build --show-bin-path`.
+
+## Using the build plugin
+
+The build plugin (`GRPCProtobufGenerator`) is a great choice for convenient dynamic code generation, however it does come with some limitations.
+Because it generates the gRPC Swift stubs as part of the build it has the requirement that `protoc` must be guaranteed
+to be available at compile time. Also because of a limitation of Swift Package Manager build plugins, the plugin
+will only be invoked when applied to the source contained in a leaf package, so it is not useful for generating code for
+library authors.
+
+The build plugin will detect `.proto` files in the source tree and perform one invocation of `protoc` for each file 
+(caching results and performing the generation as necessary).
+
+### Adoption
+Swift Package Manager build plugins must be adopted on a per-target basis, you can do this by modifying your 
+package manifest (`Package.swift` file). You will need to declare the `grpc-swift-protobuf` package as a package
+dependency and then add the plugin to any desired targets.
+
+For example, to make use of the plugin for generating gRPC Swift stubs as part of the 
+`plugin-adopter` target:
+```swift
+targets: [
+   .executableTarget(
+     name: "plugin-adopter",
+     dependencies: [
+       // ...
+     ],
+     plugins: [
+       .plugin(name: "GRPCProtobufGenerator", package: "grpc-swift-protobuf")
+     ]
+   )
+ ]
+```
+Once this is done you need to ensure that that the `.proto` files to be used for generation 
+are included in the target's source directory (below relevant the `Source` directory) and that you have 
+defined at least one configuration file.
+
+### Configuration
+
+The build plugin requires a configuration file to be present in a directory which encloses all `.proto` files 
+(in the same directory or a parent).
+Configuration files are JSON which tells the build plugin about the options which will be used in the
+invocations of `protoc`. Configuration files must be named `grpc-swift-proto-generator-config.json`
+and have the following format:
+```json
+{
+  "generate": {
+    "clients": true,
+    "servers": true,
+    "messages": true,
+  },
+  "generatedSource": {
+    "accessLevelOnImports": false,
```

Comment:
Forgot to mention that this change never actually got made, the config is still using `useAccessLevelOnImports`. I can fix that up though.

Reply:
https://github.com/grpc/grpc-swift-protobuf/pull/35